parlai/tasks/ccpe/agents.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import FixedDialogTeacher
+from .build import build
+import os
+import json
+
+
+class CCPEAllTeacher(FixedDialogTeacher):
+    def __init__(self, opt, shared=None):
+        # store datatype
+        super().__init__(opt, shared)
+
+        dt = opt['datatype'].split(':')[0]
+        if dt != 'train':
+            raise RuntimeError('Not valid datatype (only train).')
+
+        if shared:
+            self.data = shared['data']
+        else:
+            build(opt)
+            self._setup_data()
+
+        self.reset()
+
+    def num_episodes(self):
+        return len(self.data)
+
+    def num_examples(self):
+        return sum([len(x) for x in self.data])
+
+    def _setup_data(self):
+
+        fpath = os.path.join(self.opt['datapath'], 'CCPE', 'ccpe.json')
+
+        with open(fpath, 'r') as infile:
```

Comment:
does loading the file like the following not work for this data? just curious

```
with open(fpath, 'r') as infile:
    json_data = json.load(infile)
```

Reply:
I tried that, but the data itself is a malformed json file with multiple json objects that are not in an array so I had to manually make an array to contain each of the objects

Reply:
ahh ok thanks, just wanted to check